client/state/jetpack-connection-health/actions.js

```diff
@@ -37,3 +37,66 @@ export const setJetpackConnectionUnhealthy = ( siteId, errorCode ) => ( {
 	siteId,
 	errorCode,
 } );
+
+/**
+ * Requests the Jetpack connection status from the server
+ *
+ * This is called when the Jetpack connection is maybe unhealthy and we want to confirm
```

Comment:
Aren't we calling it regardless of whether it's healthy or not? We don't know if it's healthy, that's why we're calling it, right?

Reply:
No, this is not the case, and I guess it's a bit confusing. There is a jetpack connection endpoint that is getting an error originally either through `fetchModuleList` or `failedJITM`.
Only then we call that API to get a better understanding of the error.
The reason we did it like this is that this API call is expensive, so we wanted to minimize when it's called and when it's not.

Reply:
Thanks for elaborating. I guess this is a good opportunity to expand that comment with what you just explained.